go/nbs/store.go

```diff
@@ -421,7 +425,14 @@ func (nbs *NomsBlockStore) updateManifest(current, last hash.Hash) error {
 
 	candidate := nbs.tables
 	var compactees chunkSources
-	if candidate.Size() > nbs.maxTables {
+
+	shouldCompact := func() bool {
+		// As the number of tables grows from 1 to maxTables, the probability of compacting, grows from 0 to 1
```

Comment:
My only question is whether we should make it so that we never compact if there's just one table, because that's nothing but wasted time